### PR TITLE
[src] added support, tests for docker swarm integration

### PR DESCRIPTION
**Pull request by alexindeed**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
 ## 1.2.0 (Unreleased)
+
+ENHANCEMENTS
```
Comment (ZachiNachshon) [inline CHANGELOG.md]: CHANGLOG file is being edited manually before being merged to Hashicorp repository.

File: spotinst/elastigroup_integrations/fields_spotinst_elastigroup_integrations_ecs.go [hunk 2]
```diff
@@ -41,6 +41,11 @@ func SetupEcs(fieldsMap map[commons.FieldName]*commons.GenericField) {
 						Optional: true,
 					},
 
+					string(AutoscaleIsAutoConfig): &schema.Schema{
```
Comment (ZachiNachshon) [inline spotinst/elastigroup_integrations/fields_spotinst_elastigroup_integrations_ecs.go]: Please create a separate PR for the autoscale is auto config.

File: website/docs/r/elastigroup_aws.html.markdown [hunk 3]
```diff
@@ -592,6 +593,41 @@ Usage:
     }
 ```
 
+* `integration_docker_swarm` - (Optional) Describes the [Docker Swarm](https://api.spotinst.com/integration-docs/elastigroup/container-management/docker-swarm/docker-swarm-integration/) integration.
+
+    * `master_host` - (Required) IP or FQDN of one of your swarm managers.
+    * `master_port` - (Required) Network port used by your swarm. (default 2377)
```
Comment (ZachiNachshon) [inline website/docs/r/elastigroup_aws.html.markdown]: The master port is a required field on our API without default value of 2377.

**Comment by alexindeed** [comment]
- separated commits to new PR
- updated website documentation

File: website/docs/r/elastigroup_aws.html.markdown [hunk 4]
```diff
@@ -592,6 +592,41 @@ Usage:
...
+            * `memory_per_unit` - (Optional, Default: 0) Total amount of memory in each headroom unit.
+            * `number_of_units` - (Optional, Default: 0) Total number of units across the cluster to keep in reserve.
+        * `down` - (Optional) Describes scale down configuration.
+            * `evaluation_periods` - (Optional, Default: 5) Number of periods over which data is compared to the specific threshold. Minimum 3.
+            
+Usage:
+
+```hcl
+integration_docker_swarm {
+    master_host:  “10.10.10.10”
```
Comment (ZachiNachshon) [inline website/docs/r/elastigroup_aws.html.markdown]: Please fix bad quotes.

File: website/docs/r/elastigroup_aws.html.markdown [hunk 5]
```diff
@@ -592,6 +592,41 @@ Usage:
...
+                cpu_per_unit:     1000000000
+                memory_per_unit:  8000000000
+                number_of_units:  3
+            }
+        down: {
+            evaluation_periods: 3
+        }
+    }
+  }
+
```
Comment (ZachiNachshon) [inline website/docs/r/elastigroup_aws.html.markdown]: Forgot to add hcl snippet closing.

File: website/docs/r/elastigroup_aws.html.markdown [hunk 6]
```diff
@@ -592,6 +592,41 @@ Usage:
...
+        * `down` - (Optional) Describes scale down configuration.
+            * `evaluation_periods` - (Optional, Default: 5) Number of periods over which data is compared to the specific threshold. Minimum 3.
+            
+Usage:
+
+```hcl
+integration_docker_swarm {
+    master_host:  “10.10.10.10”
+    master_port:  1234
+    autoscale: {
```
Comment (ZachiNachshon) [inline website/docs/r/elastigroup_aws.html.markdown]: Illegal TF HCL, please use proper structure.

File: website/docs/r/elastigroup_aws.html.markdown (hunk 4 above)
Comment (alexindeed) [reply website/docs/r/elastigroup_aws.html.markdown]: Comparing to nomad:
```
```hcl
  integration_nomad = {
    master_host          = "my-nomad-host"
```
Are you referring to line 613? If so, this is consistent with the rest of the documentation (after aligning it with hcl formatting)

File: website/docs/r/elastigroup_aws.html.markdown (hunk 5 above)
Comment (alexindeed) [reply website/docs/r/elastigroup_aws.html.markdown]: fixed

File: website/docs/r/elastigroup_aws.html.markdown (hunk 6 above)
Comment (alexindeed) [reply website/docs/r/elastigroup_aws.html.markdown]: corrected

File: website/docs/r/elastigroup_aws.html.markdown [hunk 7]
```diff
@@ -592,6 +592,44 @@ Usage:
...
+            * `cpu_per_unit` - (Optional, Default: `0`) Total amount of CPU units in each headroom unit.
+            * `memory_per_unit` - (Optional, Default: `0`) Total amount of memory in each headroom unit.
+            * `number_of_units` - (Optional, Default: `0`) Total number of units across the cluster to keep in reserve.
+        * `down` - (Optional) Describes scale down configuration.
+            * `evaluation_periods` - (Optional, Default: `5`) Number of periods over which data is compared to the specific threshold. Minimum 3.
+            
+Usage:
+
+```hcl
+integration_docker_swarm = {
```
[the remark this answers is not in the document]
Comment (alexindeed) [reply website/docs/r/elastigroup_aws.html.markdown]: fixed

**Comment by alexindeed** [comment]
All requested changes made.

File: website/docs/r/elastigroup_aws.html.markdown [hunk 8]
```diff
@@ -592,6 +592,44 @@ Usage:
...
+    master_host = "10.10.10.10"
+    master_port = 1234
+    
+    autoscale = {
+        is_enabled: = true
```
Comment (ZachiNachshon) [inline website/docs/r/elastigroup_aws.html.markdown]: Please fix.

**Pull request merged**